packages/js-lib/src/json-schema/jsonSchemaBuilder.ts

```diff
@@ -58,9 +61,38 @@ export const j = {
       $ref,
     })
   },
-  enum<T = unknown>(enumValues: T[]) {
-    return new JsonSchemaAnyBuilder<T, JsonSchemaEnum<T>>({ enum: enumValues })
+
+  enum<const T extends readonly (string | number | boolean | null)[] | StringEnum | NumberEnum>(
+    values: T,
```

Comment:
p3: to name it `input`, since we don't know if it's an array, or object, or smth yet